lib/rest-api.php

```diff
@@ -18,6 +18,23 @@
 function gutenberg_register_rest_routes() {
 	$controller = new WP_REST_Block_Renderer_Controller();
 	$controller->register_routes();
+
+	foreach ( get_post_types( array( 'show_in_rest' => true ), 'objects' ) as $post_type ) {
+		$class = ! empty( $post_type->rest_controller_class ) ? $post_type->rest_controller_class : 'WP_REST_Posts_Controller';
+
+		if ( ! class_exists( $class ) ) {
+			continue;
+		}
+		$controller = new $class( $post_type->name );
+		if ( ! is_subclass_of( $controller, 'WP_REST_Controller' ) ) {
```

Comment:
What purpose is this condition serving? To avoid registering the autosave route if the post type REST controller class is invalid? Can we leave a remark as an inline code comment?

Also, do we need to instantiate the class, or just pass the class name to `is_subclass_of` ?

```php
<?php

if ( ! is_subclass_of( $class, 'WP_REST_Controller' ) ) {
	continue;
}
```

http://php.net/manual/en/function.is-subclass-of.php

If we want to go one step further, we could take advantage of the fact that this function returns `false` if the class doesn't exist, so we could even eliminate the prior `class_exists` condition.

```
$ php -a
php > var_export( is_subclass_of( 'Foo', 'Bar' ) );
false
```